### PR TITLE
[WIP] SameDiff TF import

### PR DESCRIPTION
**Pull request by AlexDBlack**

### CODE REVIEW

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/broadcast/BiasAddGrad.java [hunk 1]
```diff
@@ -67,7 +68,8 @@ public String onnxName() {
 
     @Override
     public String tensorflowName() {
-        return "BiasAddGrad";
+        throw new NoOpNameFoundException("TF mapping is skipped for " + opName());
+        //return "BiasAddGrad";
     }
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/broadcast/BiasAddGrad.java]: We can just delete the whole tensorflowName method here, DynamicCustomOp tensorflowName() method throws an exception already.
Same thing for all the other classes like this.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/image/CropAndResize.java [hunk 2]
```diff
@@ -89,8 +91,8 @@ public void initFromTensorFlow(NodeDef nodeDef, SameDiff initWith, Map<String, A
     }
 
     protected void addArgs() {
-        iArguments.clear();
-        tArguments.clear();
+        //iArguments.clear();
+        //tArguments.clear();
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/image/CropAndResize.java]: Checking code, it should be safe to remove this. Let's actually delete it then :)

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/scalar/comparison/ScalarLessThan.java [hunk 3]
```diff
@@ -65,7 +65,7 @@ public String onnxName() {
 
     @Override
     public String tensorflowName() {
-        return "less";
+        return "Less";
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/scalar/comparison/ScalarLessThan.java]: This mapping is incorrect, and already mapped to LessThan custom op - remove method.

**Review by AlexDBlack — COMMENTED** [review]
Overall LGTM, few minor issues to fix.
Main thing is a cleanup for tensorflowName() / onnxName() methods (we can just delete them).

After that, it can be merged :+1:

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/broadcast/BiasAddGrad.java (hunk 1 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/broadcast/BiasAddGrad.java]: Fixed for DynamicCustomOp  children.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/image/CropAndResize.java (hunk 2 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/image/CropAndResize.java]: Fixed.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/scalar/comparison/ScalarLessThan.java (hunk 3 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/scalar/comparison/ScalarLessThan.java]: Fixed.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/convolution/Pooling2D.java [hunk 4]
```diff
@@ -233,7 +234,8 @@ public String onnxName() {
 
     @Override
     public String tensorflowName() {
-        return "Pooling2D";
+        throw new NoOpNameFoundException("Tensorflow name not found for " + opName());
+        //return "Pooling2D";
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/convolution/Pooling2D.java]: Remove

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/loss/SigmoidCrossEntropyLoss.java [hunk 5]
```diff
@@ -66,7 +66,8 @@ public String opName() {
 
     @Override
     public String tensorflowName() {
-        return "sigmoid_cross_entropy";
+        throw new NoOpNameFoundException("No tensorflow name found for " + opName());
+        //return "sigmoid_cross_entropy";
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/loss/SigmoidCrossEntropyLoss.java]: Remove

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/reduce/bool/IsInf.java [hunk 6]
```diff
@@ -66,7 +67,8 @@ public String onnxName() {
 
     @Override
     public String tensorflowName() {
-        return "HasInf";
+        throw new NoOpNameFoundException("No tensorflow name found for " + opName());
+        //return "HasInf";
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/reduce/bool/IsInf.java]: Let's remove this (and other reduce bool ops here) and instead put a method with exception in BaseOp - for tensorflowName() and onnxName() - same as DynamicCustomOp

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/reduce/floating/Entropy.java [hunk 7]
```diff
@@ -64,7 +64,8 @@ public String onnxName() {
 
     @Override
     public String tensorflowName() {
-        return "entropy";
+        throw new NoOpNameFoundException("No tensorflow op opName found for " +  opName());
+        //return "entropy";
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/reduce/floating/Entropy.java]: As per the reduce bool ops - move to BaseOp

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/scalar/comparison/ScalarLessThan.java [hunk 8]
```diff
@@ -65,7 +65,7 @@ public String onnxName() {
 
     @Override
     public String tensorflowName() {
-        return "less";
+        return "LessThan";
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/scalar/comparison/ScalarLessThan.java]: This is still wrong.
Previous comment:
> This mapping is incorrect, and already mapped to LessThan custom op - remove method.

By this I meant that ScalarLessThan should not have a tensorflowName() method at all.
The previous "Less" TF name was incorrect, and was already imported in the ND4J LessThan method: https://github.com/eclipse/deeplearning4j/blob/master/nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/transforms/custom/LessThan.java#L66

So to be clear, delete this tensorflowName() method for the ScalarLessThan op.

**Review by AlexDBlack — COMMENTED** [review]
There's still a whole lot of tensorflowName methods that should be removed. To do that for these methods (that don't extend DynamicCustomOp), we'll need a tensorflowName method (that throws an exception) in BaseOp. Also an onnxName() method there too.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/loss/SigmoidCrossEntropyLoss.java (hunk 5 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/loss/SigmoidCrossEntropyLoss.java]: Done

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/convolution/Pooling2D.java (hunk 4 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/convolution/Pooling2D.java]: Done

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/reduce/bool/IsInf.java (hunk 6 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/reduce/bool/IsInf.java]: Done

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/scalar/comparison/ScalarLessThan.java (hunk 8 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/scalar/comparison/ScalarLessThan.java]: Done

**Pull request merged**